pkg/sql/catalog/descs/descriptor.go

```diff
@@ -163,20 +154,14 @@ func (tc *Collection) getByName(
 	avoidCached, mutable bool,
 ) (found bool, desc catalog.Descriptor, err error) {
 
-	// Handle special specific behaviors for the implied type.
 	var parentID, parentSchemaID descpb.ID
-	switch {
-	case db == nil && sc == nil: // database
-		if db := maybeGetSystemDatabase(name, mutable); db != nil {
-			return true, db, nil
```

Comment:
The logic for handling the special case of the system database now lives in `kv_descriptor.go`.